Core/tests/WholeProjectRefactorer.cpp

```diff
@@ -31,134 +31,536 @@
+  // TestEvent: ObjectAction
   {
-    auto &layout = project.InsertNewLayout("LayoutWithFreeFunctions", 0);
-    auto &externalEvents =
-        project.InsertNewExternalEvents("ExternalEventsWithFreeFunctions", 0);
-    externalEvents.SetAssociatedLayout("LayoutWithFreeFunctions");
+    gd::StandardEvent event;
+    gd::Instruction instruction;
+    instruction.SetType(
+        "MyEventsExtension::MyEventsBasedObject::MyObjectEventsFunction");
+    instruction.SetParametersCount(4);
+    instruction.SetParameter(0, gd::Expression("MyCustomObject"));
+    instruction.SetParameter(1, gd::Expression("First parameter"));
+    instruction.SetParameter(2, gd::Expression("Second parameter"));
+    instruction.SetParameter(3, gd::Expression("Third parameter"));
+    event.GetActions().Insert(instruction);
+    eventList.InsertEvent(event);
+  }
 
-    // Create an event in the layout referring to
-    // MyEventsExtension::MyEventsFunction
-    {
-      gd::StandardEvent event;
-      gd::Instruction instruction;
-      instruction.SetType("MyEventsExtension::MyEventsFunction");
-      instruction.SetParametersCount(4);
-      instruction.SetParameter(0, gd::Expression("scene"));
-      instruction.SetParameter(1, gd::Expression("First parameter"));
-      instruction.SetParameter(2, gd::Expression("Second parameter"));
-      instruction.SetParameter(3, gd::Expression("Third parameter"));
-      event.GetActions().Insert(instruction);
-      layout.GetEvents().InsertEvent(event);
-    }
+  // Create an event in the layout using "MyProperty" action
+  // TestEvent: ObjectPropertyAction
+  {
+    gd::StandardEvent event;
+    gd::Instruction instruction;
+    instruction.SetType(
+        "MyEventsExtension::MyEventsBasedObject::" +
+        gd::EventsBasedObject::GetPropertyActionName("MyProperty"));
+    event.GetActions().Insert(instruction);
+    eventList.InsertEvent(event);
+  }
 
-    // Create an event in the external events referring to
-    // MyEventsExtension::MyEventsFunctionExpression
-    {
-      gd::StandardEvent event;
-      gd::Instruction action;
-      action.SetType("MyExtension::DoSomething");
-      action.SetParametersCount(1);
-      action.SetParameter(
-          0,
-          gd::Expression(
-              "1 + MyEventsExtension::MyEventsFunctionExpression(123, 456)"));
-      event.GetActions().Insert(action);
-      externalEvents.GetEvents().InsertEvent(event);
-    }
+  // Create an event in the layout using "MyProperty" condition
+  // TestEvent: ObjectPropertyCondition
+  {
+    gd::StandardEvent event;
+    gd::Instruction instruction;
+    instruction.SetType(
+        "MyEventsExtension::MyEventsBasedObject::" +
+        gd::EventsBasedObject::GetPropertyConditionName("MyProperty"));
+    event.GetConditions().Insert(instruction);
+    eventList.InsertEvent(event);
+  }
+
+  // Create an event in the layout using "MyProperty" expression
+  // TestEvent: ObjectPropertyExpression
```

Comment:
Any chance you could somehow detect the position of the inserted event, and if the `position != ObjectPropertyExpression`, then throw (because the set up would be done incorrectly, so the tests will fail for sure).